src/global/signal_handler.cc

```diff
@@ -313,13 +313,33 @@ struct SignalHandler : public Thread {
 	    r = read(handlers[signum]->pipefd[0], &v, 1);
 	    if (r == 1) {
 	      siginfo_t * siginfo = &handlers[signum]->info_t;
-	      string task_name = get_name_by_pid(siginfo->si_pid);
-	      derr << "received  signal: " << sig_str(signum)
-		   << " from " << " PID: " << siginfo->si_pid
-		   << " task name: " << task_name
-		   << " UID: " << siginfo->si_uid
-		   << dendl;
-	      handlers[signum]->handler(signum);
+              ostringstream message;
+              message << "received  signal: " << sig_str(signum);
+              switch (siginfo->si_code) {
+                case SI_USER:
+                  message << " from " << get_name_by_pid(siginfo->si_pid);
+                  // If PID is undefined, it doesn't have a meaning to be displayed
+                  if (siginfo->si_pid) {
+                    message << " (PID: " << siginfo->si_pid << ")";
+                  } else {
+                    message << " ( Could be generated by pthread_kill(), raise(), abort(), alarm() )";
+                  }
+                  message << " UID: " << siginfo->si_uid;
+                  break;
+                default:
+                  /* As we have an unhandled signal, let's report the structure to help debugging */
+                  message << ", si_code : " << siginfo->si_code;
+                  message << ", si_value (int): " << siginfo->si_value.sival_int;
+                  message << ", si_value (ptr): " << siginfo->si_value.sival_ptr;
+                  message << ", si_errno: " << siginfo->si_errno;
+                  message << ", si_pid : " << siginfo->si_pid;
+                  message << ", si_uid : " << siginfo->si_uid;
+                  message << ", si_addr" << siginfo->si_addr;
+                  message << ", si_status" << siginfo->si_status;
```

Comment:
i'd suggest not print these variables unless they will be helpful for postmortem debugging.

Reply:
That's exactly the purpose of that part. When I had to debug a issue related to this code, I had to print all that structure as fields depends on the signaled as per : https://www.mkssoftware.com/docs/man5/siginfo_t.5.asp#Signal_Codes

Reply:
@ErwanAliasr1 i don't really understand. if your intention is to debugging an issue related to this code, i guess you've finished debugging, no?

Reply:
Looks like I'm not clear enough :)
During my debugging sessions, If I had that information being printed, I would have been in a better position to understand what type of signal we got & its meaning.
So my debbugging sessions was to print that information to sort out the case we were in, meaning, checkouting the tag, adding this patch, rebuilding & trick the versionning to let this binary being accepted by the other libs.
That's a lot of energy for very few information.
This code isn't supposed to be printed unless we are in a weird case. If so, printing will save a lot of time for the support/developper.